beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs

```diff
@@ -345,7 +331,17 @@ impl<E: EthSpec> PendingComponents<E> {
             block,
             blobs: Some(verified_blobs),
             blobs_available_timestamp,
-            data_columns: Some(verified_data_columns),
+            // TODO(das) Do we need a check here for number of expected custody columns?
+            // TODO(das): Update store types to prevent this conversion
+            data_columns: Some(
+                VariableList::new(
+                    verified_data_columns
+                        .into_iter()
+                        .map(|d| d.into_inner())
+                        .collect(),
+                )
+                .expect("data column list is within bounds"),
```

Comment:
This would be an unexpected scenario due to code error. Agree we should update the store types! 

I think we can `?` this to handle unexpected errors consistently.

Reply:
Hopefully we can update the store types today or tomorrow :)